ui/src/components/BeamlineCamera/BeamlineCamera.jsx

```diff
@@ -1,127 +1,129 @@
+                        }
+                        size="sm"
+                      >
+                        <img src={pip} alt="PIP Icon" />
+                      </Button>
+                    </div>
+                    <div className="vr" />
+                    <div>
+                      <MdClose
+                        color="red"
+                        onClick={() => handleShowVideoModal(key, false)}
+                        size="1.5em"
+                        className={styles.closeBtn}
+                      />
+                    </div>
+                  </Stack>
+                </Card.Header>
+                <Card.Body>
+                  {camera.format !== 'mp4' ? (
+                    <img
+                      src={camera.url}
+                      alt={camera.label}
+                      width={camera.width}
+                      height={camera.height}
+                    />
+                  ) : (
+                    <video
+                      src={camera.url}
+                      alt={camera.label}
+                      width={camera.width}
+                      height={camera.height}
+                    />
+                  )}
+                </Card.Body>
+              </Card>
+            </Draggable>
+          </div>
+        ) : null,
       );
     }
+    return DraggableElements;
+  }
 
-    return (
-      <OverlayTrigger
-        show={showLabelOverlay}
-        rootClose
-        trigger="click"
-        placement="bottom"
-        overlay={optionsOverlay}
-      >
-        <div onClick={() => setShowLabelOverlay(!showLabelOverlay)}>
-          <Badge
-            bg="secondary"
-            style={{ display: 'block', marginBottom: '3px' }}
-          >
-            {labelText}
-            <i className="fas fa-cog ms-2" />
-          </Badge>
-        </div>
-      </OverlayTrigger>
-    );
-  };
-
-  const msgLabelStyle = {
-    display: 'block',
-    fontSize: '100%',
-    borderRadius: '0px',
-    color: '#000',
-  };
+  function renderCamera() {
+    const dropdownItem = [];
+    for (const [key, camera] of cameraSetup.components.entries()) {
+      dropdownItem.push(
+        <Dropdown.Item
+          key={key}
+          onClick={() => handleShowVideoModal(key, true)}
+        >
+          {camera.label} <i className="fas fa-video" />
+        </Dropdown.Item>,
+        <Dropdown.Divider />,
+      );
+    }
+    return dropdownItem;
+  }
 
-  const video = (
-    <div>
-      {format !== 'mp4' ? (
-        <img // eslint-disable-line jsx-a11y/no-noninteractive-element-interactions
-          onClick={onImageClick}
-          src={url}
-          alt={labelText}
-          width={width}
-          height={height}
-        />
-      ) : (
-        <video
-          src={url}
-          alt={labelText}
-          onClick={onImageClick}
-          width={width}
-          height={height}
-        />
-      )}
-      <Button
+  return cameraSetup?.components.length > 0 ? (
```

Comment:
You can move this into an early return:

```jsx
if (!cameraSetup?.components.length) {
  return null;
}

return (<>/* ... */</>);
```

Reply:
Actually, it looks like all you care about is `cameraSetup.components`, so this array should probably become the prop and the `<BeamlineCamera>` component should be rendered conditionally in `BeamlineSetupContainer` — something like:

```jsx
const { components = [] } = uiproperties.camera_setup || {};

{components.length > 0 && <BeamlineCamera components={components} />}
```